backend/wallet/services/offchain/payment.py

```diff
@@ -0,0 +1,234 @@
+        source_id=account_id,
+        source_address=my_address,
+        source_subaddress=my_sub_address,
+        destination_id=None,
+        destination_address=vasp_address,
+        destination_subaddress=vasp_sub_address,
+        reference_id=reference_id,
+    )
+
+
+def send_init_auth_payment_request(payment_model, account_id):
+    user = storage.get_user(account_id)
+
+    try:
+        utils.offchain_client().send_request(
+            request_sender_address=payment_model.my_address,
+            counterparty_account_id=payment_model.vasp_address,
+            request_bytes=jws.serialize(
+                new_init_auth_command(
+                    reference_id=payment_model.reference_id,
+                    vasp_address=payment_model.vasp_address,
+                    sender_name=user.first_name,
+                    sender_sure_name=user.last_name,
+                    sender_city=user.city,
+                    sender_country=user.country,
+                    sender_line1=user.address_1,
+                    sender_line2=user.address_2,
+                    sender_postal_code=user.zip,
+                    sender_state=user.state,
+                    sender_national_id_value="",
+                    sender_national_id_type="",
+                ),
+                utils.compliance_private_key().sign,
+            ),
+        )
+    except Exception as e:
+        error = P2MGeneralError(e)
+        logger.error(error)
+        raise error
+
+
+def send_init_charge_payment_request(payment_model, account_id):
+    user = storage.get_user(account_id)
+
+    try:
+        command_response_object = utils.offchain_client().send_request(
+            request_sender_address=payment_model.my_address,
+            counterparty_account_id=payment_model.vasp_address,
+            request_bytes=jws.serialize(
+                new_init_charge_command(
+                    reference_id=payment_model.reference_id,
+                    vasp_address=payment_model.vasp_address,
+                    sender_name=user.first_name,
+                    sender_sure_name=user.last_name,
+                    sender_city=user.city,
+                    sender_country=user.country,
+                    sender_line1=user.address_1,
+                    sender_line2=user.address_2,
+                    sender_postal_code=user.zip,
+                    sender_state=user.state,
+                    sender_national_id_value="",
+                    sender_national_id_type="",
+                ),
+                utils.compliance_private_key().sign,
+            ),
+        )
+
+        if (
+            command_response_object.result
+            and type(command_response_object.result) is InitChargePaymentResponse
+        ):
+            recipient_signature = command_response_object.result.recipient_signature
+
+            storage.update_payment(payment_model.reference_id, recipient_signature)
+    except Exception as e:
```

Comment:
This is very discouraged. Is it possible to know the specific exception type? And why would you catch it at all if you do nothing with it? Won't the outer layers handle it?

Reply:
as my previous answer